core/common/src/test/scala/org/typelevel/otel4s/TextMapGetterProps.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+
+import munit.ScalaCheckSuite
+import org.scalacheck.Arbitrary
+import org.scalacheck.Prop.forAll
+
+import scala.collection.concurrent
+import scala.collection.generic.IsIterable
+import scala.collection.immutable
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+class TextMapGetterProps extends ScalaCheckSuite {
+  def mkProps[A: Arbitrary](implicit
+      tag: ClassTag[A],
+      getter: TextMapGetter[A],
+      conv: IsIterable[A] { type A = (String, String) }
+  ): Unit = {
+    val className =
+      tag.runtimeClass match {
+        case cls if cls.isArray => "Array"
+        case cls                => cls.getName
+      }
+    property(s"TextMapSetter#get  for $className") {
+      forAll { (coll: A, key: String) =>
+        assertEquals(
+          getter.get(coll, key),
+          conv(coll).find(_._1 == key).map(_._2)
```

Comment:
This is quietly reiterating that first one wins with duplicates, but I think we covered ourselves with the doc.

Reply:
yeah... I'm not sure how I feel about it. I might be able to restructure it so that it doesn't? perhaps something like

```scala
val maybeValue = getter.get(coll, key)
assertEquals(maybeValue.isDefined, conv(coll).exists(_._1 == key))
if (maybeValue.isDefined) assert(conv(coll).exists(_ == (key -> maybeValue.get)))
```

feels ugly to me though. the simple answer as to why I wrote the property that way is "it was easier"

Reply:
slightly nicer I think:

```scala
      forAll { (coll: A, key: String) =>
        val res = getter.get(coll, key)
        assertEquals(res.isDefined, conv(coll).exists(_._1 == key))
        assert(res.forall(value => conv(coll).exists(_ == (key -> value))))
      }
```